File: solutions/lua/resistor-color-duo/1/resistor-color-duo.lua
local COLORCODES = {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}

return {
  value = function(colors)
    if #colors < 2 then
      error(string.format("value() expects an array of length 2 or longer, received length %d", #colors))
    end
    return tonumber(string.format("%d%d", COLORCODES[colors[1]], COLORCODES[colors[2]]))
  end
}

File: solutions/typescript/resistor-color-duo/4/resistor-color-duo.ts
const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
] as const;

export type Color = typeof COLORS[number];

export const decodedValue = (colors:Color[]) => {
  if (colors.length < 2) {
    throw new Error("At least two colors are required");
  }
  const firstColorValue = COLORS.indexOf(colors[0]);
  const secondColorValue = COLORS.indexOf(colors[1]);

  return firstColorValue * 10 + secondColorValue;
};